draft-ietf-tls-exported-authenticator.md

```diff
@@ -511,6 +511,13 @@ The signatures generated with this API cover the context string
 "Exported Authenticator" and therefore cannot be transplanted into other
 protocols.
 
+In TLS 1.3 the client and server are not guaranteed to agree on the client’s
+final flight until the first application message.  Because EAs can be
```

Comment:
As noted, this first is false.  0-RTT or 0.5-RTT both violate this.

Reply:
We do get a guarantee of agreement eventually, though nailing down a concise way to describe exactly when that happens may take some thought.  IIRC it's successful use of application_traffic_secret_0.
I think the phrasing would be a little awkward if we just jump straight in as "Because EAs can be negotiated out-of-band it is possible to negotiate EAs without agreeing on the entire transcript".  If we don't want to mention the lack of transcript agreement at all, we might flip it around, "It is possible to negotiate EAs without agreeing on the entire transcript, as might occur if [...]".  We might also make the first sentence blander without removing it entirely, to mention that the non-agreement on the client's final flight is possible/a risk, without making a specific claim of when the risk goes away.

Reply:
I don't believe that that's true. application_traffic_secret_0 does not include anything from the client's last flight.

Reply:
What do people think of this text instead:
```
In TLS 1.3 the client does not explicitly learn whether its Finished message
was accepted. Because the application traffic keys are not dependent on the
clients final flight, the client cannot learn whether the server ever received
it.  To avoid disagreement between the client and server on the authentication
status of EAs servers MUST verify the client Finished before sending an EA.
```

Reply:
That seems like a good approach.  I might spend a few words for "client cannot learn just from TLS messages" or similar, since some application protocols on top of TLS might enable the client to do so.  But I don't think it's strictly necessary to have such a clarification.